core/player/src/view/plugins/switch.ts

```diff
@@ -33,75 +47,92 @@ export default class SwitchPlugin implements ViewPlugin {
-      ) {
-        return NodeType.Switch;
-      }
-    });
-
     parser.hooks.parseNode.tap(
       "switch",
       (
         obj: any,
         _nodeType: Node.ChildrenTypes,
         options: ParseObjectOptions,
-        determinedNodeType: null | NodeType,
+        childOptions?: ParseObjectChildOptions,
       ) => {
-        if (determinedNodeType === NodeType.Switch) {
-          const dynamic = "dynamicSwitch" in obj;
-          const switchContent =
-            "dynamicSwitch" in obj ? obj.dynamicSwitch : obj.staticSwitch;
-
-          const cases: Node.SwitchCase[] = [];
-
-          switchContent.forEach(
-            (switchCase: {
-              [x: string]: any;
-              /**
-               *
-               */
-              case: any;
-            }) => {
-              const { case: switchCaseExpr, ...switchBody } = switchCase;
-              const value = parser.parseObject(
-                switchBody,
-                NodeType.Value,
-                options,
-              );
-
-              if (value) {
-                cases.push({
-                  case: switchCaseExpr,
-                  value: value as Node.Value,
-                });
-              }
-            },
-          );
-
-          const switchAST = parser.hooks.onCreateASTNode.call(
+        if (
+          this.isSwitch(obj) ||
+          (childOptions && hasSwitchKey(childOptions.key))
+        ) {
+          const objToParse =
+            childOptions && hasSwitchKey(childOptions.key)
+              ? { [childOptions.key]: obj }
+              : obj;
+          const dynamic = "dynamicSwitch" in objToParse;
+          const switchContent = dynamic
+            ? objToParse.dynamicSwitch
+            : objToParse.staticSwitch;
+
+          const cases: Node.SwitchCase[] = switchContent
+            .map(
+              (switchCase: {
+                [x: string]: any;
+                /**
+                 *
+                 */
+                case: any;
+              }) => {
+                const { case: switchCaseExpr, ...switchBody } = switchCase;
+                const value = parser.parseObject(
+                  switchBody,
+                  NodeType.Value,
+                  options,
+                );
+
+                if (value) {
+                  return {
+                    case: switchCaseExpr,
+                    value: value as Node.Value,
+                  };
+                }
+
+                return;
+              },
+            )
+            .filter(Boolean);
+
+          const switchAST = parser.createASTNode(
             {
               type: NodeType.Switch,
               dynamic,
               cases,
             },
-            obj,
+            objToParse,
           );
 
-          if (switchAST?.type === NodeType.Switch) {
+          if (!switchAST || switchAST.type === NodeType.Empty) {
+            return childOptions ? [] : null;
```

Comment:
distinction between this `null` and [this](https://github.com/player-ui/player/pull/413/files#diff-f177d6cbe60df2262c39019c6c0a9323e88dd6ddfe776691c54cefa008bee645R70) `undefined`?